src/components/homepage/EditorHeroSection/EditorHeroSection.jsx

```diff
@@ -0,0 +1,119 @@
+import { CardContainer } from "components/CardContainer"
+import { FormContext, FormError, FormTitle } from "components/Form"
+import FormField from "components/FormField"
+import FormFieldMedia from "components/FormFieldMedia"
+import { HeroDropdown, HeroHighlight } from "components/homepage/HeroSection"
+import _ from "lodash"
+import PropTypes from "prop-types"
+import React, { useEffect } from "react"
+import { useFormContext } from "react-hook-form"
+
+import elementStyles from "styles/isomer-cms/Elements.module.scss"
+
+const HeroComponent = ({ heroType, fieldId }) => {
+  if (heroType === "highlights") return <HeroHighlight fieldId={fieldId} />
+  if (heroType === "dropdown")
+    return <HeroDropdown fieldId={`${fieldId}.dropdown`} />
+  return null
+}
+
+export const EditorHeroSection = ({
+  fieldId, //  This fieldId refers to sections.0.hero, and it's a string reference id to the object in useForm
+}) => {
+  const {
+    register,
+    unregister,
+    formState: { errors },
+    setValue,
+    watch,
+  } = useFormContext()
+  const sectionErrors = _.get(errors, fieldId)
+  const heroType = watch(`${fieldId}.heroType`)
+
+  useEffect(() => {
+    // reset highlight and dropdown toggle states
+    if (heroType !== "highlights") {
+      setValue(`${fieldId}.key_highlights`, [])
+      unregister(`${fieldId}.button`)
+      unregister(`${fieldId}.url`)
+    }
+    if (heroType !== "dropdown") {
+      unregister(`${fieldId}.dropdown`)
+    }
+  }, [heroType, unregister, setValue])
```

Comment:
could i ask where are the `${fieldId}.<component>` coming from? they don't seem to be within the file itself ._.